app.rb

```diff
@@ -0,0 +1,124 @@
+require './person'
+require './student'
+require './teacher'
+require './class_room'
+require './book'
+require './rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    @classroom = Classroom.new('Grade 10')
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts 'Oops! the library is empty, please add books'
+      return
+    end
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'Oops! no people registered, please add people'
+    else
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher(2)?....'
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input! Please type 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Name:'
+    name = gets.chomp
+
+    print 'Age:'
+    age = gets.chomp.to_i
+
+    print 'Has Parent permission? [Y/N]:'
+    parent_permission = gets.chomp.downcase
+
+    student = Student.new(age, @classroom, name, parent_permission: parent_permission == 'y')
+    @people << student
+    puts 'Student created successfully'
+  end
+
+  def create_teacher
+    print 'Name:'
+    name = gets.chomp
+
+    print 'Age:'
+    age = gets.chomp.to_i
+
+    print 'Specialization:'
+    specialization = gets.chomp
+
+    @people << Teacher.new(age, specialization, name)
+    puts 'Teacher created successfully'
+  end
```

Comment:
- [ ] The same happens when I try to create a Teacher:

![image](https://github.com/lengushuru/OOP-school-library/assets/77069011/5c4b80f5-ca63-4d55-807c-25f0ef91df48)

Please consider looking into it! 🤓 🔧 

💡 _Pay special attention to how your `Teacher` class expects the parameters to be passed and how you're initializing them. See [named parameters](https://thoughtbot.com/blog/ruby-2-keyword-arguments#required-keyword-arguments)_